### DISCUSSION TITLE
Additional Data in UsernamePasswordAuthrticationrequest

### DISCUSSION THREAD

**Discussion by ckosmowski** [quarkusio/quarkus#38617] [Q&A] [answered]
Hi, I am using form based authentication with quarkus. Beside the username and password id like to pass an additional language for the login.

The value ist sent alongside the username and password data already. But I can't find a way to access the information inside the identity provider. I only have the username password authentication request object which does not seem to contain that information.

**Comment by sberyozkin** [comment]
@ckosmowski Hi, try passing this request object to `io.quarkus.vertx.http.runtime.security.HttpSecurityUtils.getRoutingContextAttribute`, the returned `RoutingContext` should let you access the extra request properties, query, headers.

**Reply by ckosmowski** [reply]
Thank you! Will try that

**Reply by ckosmowski** [reply]
I tried that solution but i cannot find the additional data there. The data contained in the routingContext looks like this:

![image](https://github.com/quarkusio/quarkus/assets/2411084/cf8f5500-eebb-491d-a490-ba29fc3f3cbc)

I already "went down" the tree of objects but i could not find the additional data.

**Reply by sberyozkin** [reply]
@ckosmowski How is this language identifier passed in the HTTP request ? As a form parameter ?

**Reply by ckosmowski** [reply]
Yes it is passed as a form parameter

**Reply by sberyozkin** [reply]
@ckosmowski Can you try this PR, #38652, and let me know if it helps ?  Thanks

**Reply by michalvavrik** [reply, marked as answer]
@ckosmowski it's not clear to me what is your use case, but so far it seems to me like you try to access the `RoutingContext` in an augmentor and it is not there, correct? Please try the Quarkus 3.7.1 and try https://quarkus.io/guides/security-customization#security-identity-customization `RoutingContext routingContext = HttpSecurityUtils.getRoutingContextAttribute(attributes);` (last example) because then you can do `MultiMap res = routingContext.request().formAttributes();`

**Reply by sberyozkin** [reply]
@michalvavrik Access to the `RoutingContext` works in the identity provider, the use case is to access the language attribute related to the name/password. But indeed, I did not realize it may already work, thanks for pointing it out. @ckosmowski Please check

**Reply by ckosmowski** [reply]
Thanks for your fast replies. `MultiMap res = routingContext.request().formAttributes();` was the missing piece. That also works without changing the version.
